### ISSUE TITLE
Wear OS collector to send data to NightScout

### ISSUE THREAD

**Issue by dubidrubi** [NightscoutFoundation/xDrip#1611]
Afaik, Wear OS xDrip apk current can transmit data using native xDrip follower mode when works as a standalone collector. But that requires a Google service installed/running on the watch. 

Would be great to have an option to send readings to NightScout from WearOS collector in case Google services are missing. A great example is Xiaomi Mi Watch - an excellent watch with affordable price with a great HW but missing Google services. 

I confirm that xDrip runs perfectly on them and standalone collector works with no issues. Only followers option is missing.

**Comment by tolot27** [comment]
> Would be great to have an option to send readings to NightScout from WearOS collector
> ...
> I confirm that xDrip runs perfectly on them and standalone collector works with no issues. Only followers option is missing.

I'm confused. Do you say the data source "Nightscout Follower" does not work at the watch or do you like to use Nightscout Uploader from the watch?

**Comment by Navid200** [comment]
@dubidrubi Please respond to questions asked by developers.

**Comment by dubidrubi** [comment]
Hey @tolot27 @Navid200. Sorry for the delay.

2nd option - I'd like to use NS uploader on WearOS, currently, it is not possible.

**Comment by Navid200** [comment]
@dubidrubi @tolot27 Does a watch that uses WearOS have direct access to the internet?
I mean we are not talking about a full Android device here, right?

**Comment by tolot27** [comment]
> @dubidrubi @tolot27 Does a watch that uses WearOS have direct access to the internet? I mean we are not talking about a full Android device here, right?

Yes, there are WearOS watches available having WiFi.

**Comment by Navid200** [comment]
I still don't get it.
If you want your watch to upload to Nightscout, isn't that something that requires a modification to your watch?
How can we change xDriip to make your watch in a stan-alone mode to behave differently than it does now?
Please explain and we can reopen.

**Comment by lostboy86** [comment]
There are many watch WearOS wit LTE module and SIM slot.

They have direct access to Internet.

**Comment by Navid200** [comment]
@lostboy86 So, a watch that uses WearOS can independently collect from a sensor with no phone present?

**Comment by lostboy86** [comment]
Yes, at now there is this options : Libre1 + Miao à Xdrip WearOS  directlyLibre2 ->  Juggluco WearOS directlyXdrip WearOS + oop2 directlyLibre3 -> Juggluco WearOS directlyJuggluco WearOS and xdrip WearOS no have ability to send data over nightscout.Juggluco smartphone and xdrip smartphone have ability to send data over nightscout Da: NavidInviato: giovedì 16 marzo 2023 21:41A: NightscoutFoundation/xDripCc: lostboy86; MentionOggetto: Re: [NightscoutFoundation/xDrip] Wear OS collector to send data to NightScout ***@***.*** So, a watch that uses WearOS can independently collect from a sensor with no phone present?—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

**Comment by lostboy86** [comment]
Also, for clarify, smartwatch with s.o. WearOS and Sim module, with internet indipendet, are out from 2015 (my old 2016 Huawaei Watch 2 Sport LTE model had internet with sim slot indipendent)

**Comment by Navid200** [comment]
I'm hoping that a developer will move WearOS out of xDrip and into a companion app.  I hope this will allow WearOS to be better supported.
In the meantime, I am going to convert this to a discussion.
